src/wp-includes/post.php

```diff
@@ -901,36 +901,58 @@ function get_post_status( $post = null ) {
 		return false;
 	}
 
-	if ( 'attachment' === $post->post_type ) {
-		if ( 'private' === $post->post_status ) {
-			return 'private';
-		}
+	$post_status = $post->post_status;
 
-		// Unattached attachments are assumed to be published.
-		if ( ( 'inherit' === $post->post_status ) && ( 0 == $post->post_parent ) ) {
-			return 'publish';
-		}
+	if (
+		'attachment' === $post->post_type &&
+		'inherit' === $post_status
+	) {
+		// Attachment permitted statuses: 'inherit', , see wp_insert_post().
 
-		// Inherit status from the parent.
-		if ( $post->post_parent && ( $post->ID != $post->post_parent ) ) {
-			$parent_post_status = get_post_status( $post->post_parent );
-			if ( 'trash' === $parent_post_status ) {
-				return get_post_meta( $post->post_parent, '_wp_trash_meta_status', true );
-			} else {
-				return $parent_post_status;
+		if (
+			0 === $post->post_parent ||
+			! get_post( $post->post_parent ) ||
+			$post->ID === $post->post_parent
+		) {
+			// Unattached attachments with inherit status are assumed to be published.
+			$post_status = 'publish';
+		} elseif (
+			$post->post_parent &&
```

Comment:
Isn't this covered in the previous conditional `0 === $post→post_parent`. Should that conditional be a simple `! $post→post_parent` to handle `'0'` `false`, etc...?

Reply:
It sure is, removed in c0b77d662c8ccf5c1c7866a91a1f44e83d26587f